cinema/views.py

```diff
@@ -56,3 +83,75 @@
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = MovieSession.objects.prefetch_related("movie", "cinema_hall", "tickets")
+        date = self.request.query_params.get("date")
+        movie = self.request.query_params.get("movie")
+
+        if date:
+            queryset = queryset.filter(show_time__date=date)
+
+        if movie:
+            queryset = queryset.filter(movie__id=movie)
+
+        if self.action == "list":
+            queryset = queryset.select_related(
+                "movie",
+                "cinema_hall"
+            ).annotate(
+                tickets_available=F(
+                    "cinema_hall__rows"
+                ) * F(
+                    "cinema_hall__seats_in_row"
+                ) - Count(
+                    "tickets"
+                )
+            )
+
+        elif self.action == "retrieve":
+            queryset = queryset.prefetch_related("movie", "cinema_hall")
+
+        return queryset.distinct()
+
+
+class TicketViewSet(viewsets.ModelViewSet):
+    queryset = Ticket.objects.all()
+    serializer_class = TicketSerializer
+
+
+class OrderPagination(PageNumberPagination):
+    page_size = 3
+    page_size_query_param = "page_size"
+    max_page_size = 20
+
+
+class OrderViewSet(viewsets.ModelViewSet):
+    queryset = Order.objects.all()
+    serializer_class = OrderSerializer
+    pagination_class = OrderPagination
+
+    def get_queryset(self):
+        queryset = Order.objects.prefetch_related(
+            "tickets__movie_session__movie",
+            "tickets__movie_session__cinema_hall"
+        )
+        queryset = queryset.filter(user=self.request.user)
+
+        if self.action == "list":
+            queryset = queryset.prefetch_related("tickets__movie_session__cinema_hall")
+        elif self.action == "retrieve":
+            queryset = queryset.prefetch_related("tickets__movie_session")
+
+        return queryset.distinct()
+
+    def perform_create(self, serializer):
+        serializer.save(user=self.request.user)
```

Comment:
The `perform_create` method in `OrderViewSet` saves the order with the current user, ensuring that the order is associated with the correct user. This is a necessary step for user-specific data handling.